superset/data/__init__.py

```diff
@@ -36,6 +36,21 @@
 misc_dash_slices = []  # slices assembled in a "Misc Chart" dashboard
 
 
+def update_slice_ids(layout_dict, slices):
+    charts = list(
+        filter(
+            lambda component: (
```

Comment:
[optional] the more pythonic way of writing this is by using a list comprehension, something like
`charts = [o for o in layout_dict.values() if isinstance(component, dict) and component['type'] == 'DASHBOARD_CHART_TYPE']`